light/client.go

```diff
@@ -1066,13 +1098,13 @@ func (c *Client) compareFirstHeaderWithWitnesses(ctx context.Context, h *types.S
 		case nil:
 			continue
 		case errConflictingHeaders:
-			c.logger.Error(fmt.Sprintf(`witness #%d has a different header. Please check primary is correct
-and remove witness. Otherwise, use a different primary`, e.WitnessIndex), "witness", c.witnesses[e.WitnessIndex])
+			c.logger.Error("witness has a different header. Please check primary is correct and remove witness." 
+			+ "Otherwise, use a different primary`,
```

Comment:
use `fmt.Sprintln` or `fmt.Sprint` rather than string concatenation, basically always. The `fmt.Sprint` functions use a buffer pool for the underlying byte slices, and string concatenation does a lot more alloc'ing, in a way that (I've seen) cause problems at scale, so its best to avoid.

Reply:
I only used string concatenation because of the linter. I can use ` instead